esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,662 @@
+    return;
+  }
+  ESP_LOGV(TAG, "Disabling wiper %" PRIu8, wiper_idx);
+  this->reg_[wiper_idx].enabled = true;
+  if (wiper_idx < 4) {
+    this->reg_[wiper_idx].terminal_hw = true;
+    this->reg_[wiper_idx].update_terminal = true;
+  }
+}
+
+bool Mcp4461Component::increase_wiper_(Mcp4461WiperIdx wiper) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_DISABLED)));
+    return false;
+  }
+  if (this->reg_[wiper_idx].wiper_lock_active) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_LOCKED)));
+    return false;
+  }
+  if (this->reg_[wiper_idx].state == 256) {
+    ESP_LOGV(TAG, "Maximum wiper level reached, further increase of wiper %" PRIu8 " prohibited", wiper_idx);
+    return false;
+  }
+  ESP_LOGV(TAG, "Increasing wiper %" PRIu8 "", wiper_idx);
+  uint8_t reg = 0;
+  uint8_t addr;
+  addr = this->get_wiper_address_(wiper_idx);
+  reg |= addr;
+  reg |= static_cast<uint8_t>(Mcp4461Commands::INCREMENT);
+  auto err = this->write(&this->address_, reg, sizeof(reg));
+  if (err != i2c::ERROR_OK) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    return false;
+  }
+  this->reg_[wiper_idx].state++;
+  return true;
+}
+
+bool Mcp4461Component::decrease_wiper_(Mcp4461WiperIdx wiper) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_DISABLED)));
+    return false;
+  }
+  if (this->reg_[wiper_idx].wiper_lock_active) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_LOCKED)));
+    return false;
+  }
+  if (this->reg_[wiper_idx].state == 0) {
+    ESP_LOGV(TAG, "Minimum wiper level reached, further decrease of wiper %" PRIu8 " prohibited", wiper_idx);
+    return false;
+  }
+  ESP_LOGV(TAG, "Decreasing wiper %" PRIu8 "", wiper_idx);
+  uint8_t reg = 0;
+  uint8_t addr;
+  addr = this->get_wiper_address_(wiper_idx);
+  reg |= addr;
+  reg |= static_cast<uint8_t>(Mcp4461Commands::DECREMENT);
+  auto err = this->write(&this->address_, reg, sizeof(reg));
+  if (err != i2c::ERROR_OK) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    return false;
+  }
+  this->reg_[wiper_idx].state--;
+  return true;
+}
+
+uint8_t Mcp4461Component::calc_terminal_connector_byte_(Mcp4461TerminalIdx terminal_connector) {
+  uint8_t i;
+  if ((static_cast<uint8_t>(terminal_connector) == 0 || static_cast<uint8_t>(terminal_connector) == 1)) {
+    i = 0;
+  } else {
+    i = 2;
+  }
```

Comment:
FWIW I would probably simplify this a bit:
```suggestion
  uint8_t i = static_cast<uint8_t>(terminal_connector) <= 1 ? 0 : 2;
```